### PR TITLE
[HOPSWORKS-1405_hc] Add base health check with retryable actions

### PR DESCRIPTION
**Pull request by kouzant**

### CODE REVIEW

File: templates/default/retryable_health_check.sh.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+MAX_ATTEMPTS=<%= node['consul']['health-check']['max-attempts'] %>
+MULTIPLIER=<%= node['consul']['health-check']['multiplier'] %>
+
+# Helper function to run a health check
+# Function will keep retrying to ping a
+# service if it's dead until it reaches
+# a maximum number of attempts
+_run_health_check () {
```
Comment (steffengr) [inline templates/default/retryable_health_check.sh.erb:10]: The idea is good but this means that a service that succeeds 1 out of x times is considered healthy. I think we should come up with a bit more careful strategy. E.g. there are more than 2 failures then fail?

Comment (kouzant) [reply templates/default/retryable_health_check.sh.erb:10]: That's what is doing. If a check returns non-zero it retries until (a) the service is alive in which case the health check succeeds or (b) MAX_ATTEMPTS has been reached and the service is declared dead.
Did I understand something wrong on your comment?

Comment (steffengr) [reply templates/default/retryable_health_check.sh.erb:10]: I meant something else but let's not over complicate it for now. I think this is fine.

Comment (kouzant) [reply templates/default/retryable_health_check.sh.erb:10]: Alright, we can come up with something more sophisticated later.

**Pull request merged**